holidays/countries/uruguay.py

```diff
@@ -10,88 +10,164 @@
 #  License: MIT (see LICENSE file)
 
 from datetime import date
+from datetime import timedelta as td
 from gettext import gettext as tr
 
-from holidays.calendars.gregorian import JAN, APR, MAY, JUN, JUL, AUG, OCT, MON
+from holidays.calendars.gregorian import JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, DEC, MON
+from holidays.constants import BANK, PUBLIC
 from holidays.holiday_base import HolidayBase
 from holidays.holiday_groups import ChristianHolidays, InternationalHolidays
 
 
 class Uruguay(HolidayBase, ChristianHolidays, InternationalHolidays):
     """
-    https://en.wikipedia.org/wiki/Public_holidays_in_Uruguay
+    References:
+    - https://en.wikipedia.org/wiki/Public_holidays_in_Uruguay
+    - [Law #6997] https://www.impo.com.uy/diariooficial/1919/10/25/2
+    - [Decree Law #9000] https://www.impo.com.uy/bases/decretos-ley/9000-1933
+    - [Decree Law #14977] https://www.impo.com.uy/bases/decretos-ley/14977-1979
+    - [Decree Law #15535] https://www.impo.com.uy/bases/decretos-ley/15535-1984
+    - [Law #16805] http://www.parlamento.gub.uy/leyes/AccesoTextoLey.asp?Ley=16805
+    - [Law #17414] http://www.parlamento.gub.uy/leyes/AccesoTextoLey.asp?Ley=17414
     """
 
     country = "UY"
     default_language = "es"
+    supported_categories = {BANK, PUBLIC}
     supported_languages = ("en_US", "es", "uk")
 
+    # Presidential Inauguration Day.
+    presidential_inauguration_day = tr("Inauguración del Presidente de la República")
+    special_public_holidays = {
```

Comment:
It doesn't look that special (every 5 years since 1985) but it'll do it for now.